devicemodel/hw/pci/npk.c

```diff
@@ -159,11 +159,11 @@ static inline uint32_t *offset2reg(uint64_t offset)
 	return reg;
 }
 
-static inline int valid_param(int m_off, int m_num)
+static inline int valid_param(uint32_t m_off, uint32_t m_num)
 {
-	/* 256-aligned, no less than 256, no overflow */
-	if (!(m_off & 0xFF) && !(m_num & 0xFF) && (m_off >> 8) > 0
-			&& (m_num >> 8) > 0 && m_off + m_num <= NPK_SW_MSTR_NUM)
+	/* 8-aligned, no less than 8, no overflow */
+	if (!(m_off & 0x7U) && !(m_num & 0x7U) && m_off > 0x7U
```

Comment:
m_off & 0x7f is integer, not Bool, so we cann't use "!" operations, which violate MISRA-C.   use == 0 or != 0...
same for others.

also, use () for m_off > 0x7U.  same for next

Reply:
Got it, I will update the patch.

Reply:
check the entire HV code that you ever wrote. We need to fix all of them --- not only the one you changed now.

Reply:
Oh, this is for DM. Then just modify the code in this patch...